x-pack/plugins/apm/public/components/shared/charts/TransactionCharts/index.tsx

```diff
@@ -140,9 +139,28 @@ export function TransactionCharts({
       </EuiFlexItem>
     );
   }
-
   const { responseTimeSeries, tpmSeries } = charts;
   const { transactionType } = urlParams;
+  const maxY = getMaxY(responseTimeSeries);
+  let formatter = getDurationFormatter(maxY);
+
+  function onToggleLegend(visibleSeries: TimeSeries[]) {
+    if (!isEmpty(visibleSeries)) {
+      // recalculate the formatter based on the max Y from the visible series
+      const maxVisibleY = getMaxY(visibleSeries);
+      formatter = getDurationFormatter(maxVisibleY);
+    }
+  }
```

Comment:
This looks possibly error-prone. How is `TransactionLineChart` component re-rendered so for instance `getResponseTimeTickFormatter` is invoked again?

Reply:
@sqren indeed the component never re-renders, but when `getResponseTimeTickFormatter` is called the reference to the `formatter` function has changed based on the new visible series.

Reply:
If the component never re-renders the changes to `formatter` will not be displayed and this won't work...
My guess is that the component _is_ re-rendering but due to another side-effect. So it works by chance.